netbox/dcim/models/cables.py

```diff
@@ -529,46 +535,66 @@ def from_origin(cls, terminations):
                 object_to_path_node(t) for t in terminations
             ])
 
-            # Step 2: Determine the attached link (Cable or WirelessLink), if any
-            link = terminations[0].link
-            if link is None and len(path) == 1:
+            # Step 2: Determine the attached links (Cable or WirelessLink), if any
+            links = [termination.link for termination in terminations if termination.link is not None]
+            if len(links) == 0 and len(path) == 1:
                 # If this is the start of the path and no link exists, return None
                 return None
-            elif link is None:
+            elif len(links) == 0:
                 # Otherwise, halt the trace if no link exists
                 break
-            assert type(link) in (Cable, WirelessLink)
+            assert all(type(link) in (Cable, WirelessLink) for link in links)
+
+            # Create set of links in path.  Cannot use list(set()) as it does this in a non-deterministic manner
+            links_path = []
+            for link in links:
+                if object_to_path_node(link) not in links_path:
```

Comment:
We just want to silently ignore it here because we aren't checking as a "we need to stop this if this condition is hit" but as a "We already have this node in the list of current links, we don't need to add it again".  More relevant when there are multiple rear-port positions in use for a specific rear-port